tests/ifs/unit/ifs_unittests.cpp

```diff
@@ -0,0 +1,377 @@
+/*
+ * Copyright 2023 Intel Corporation.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#if defined(__x86_64__) && defined(__linux__)
+
+#include "gtest/gtest.h"
+#include "sandstone.h"
+
+#undef log_warning
+#define log_warning
+#undef log_error
+#define log_error
+#undef log_info
+#define log_info
+#undef log_debug
+#define log_debug
+
+#include "../sandstone_ifs.c"
+
+#undef PATH_SYS_IFS_BASE
+#define PATH_SYS_IFS_BASE
+
+#include "../ifs.c"
+#include "ifs_test_cases.h"
+#include "ifs_unit_utils.h"
+
+/*
+ * @brief Setup all structs, directory and files needed by test
+ */
+void *test_setup(ifs_unit setup_data)
```

Comment:
Wondering, is there a reason for which `ifs_unit` is passed as argument instead of `ifs_unit *`?

Reply:
Not really, if you think its important I can do the change.